File: RepoTransBench/target_projects/Java/Go/pilgr_Paper/public_tests/paperdb_exception_public_test.go
package public_tests

import (
	"errors"
	"testing"
)

type PaperDbException struct {
	message string
	cause   error
}

func (e *PaperDbException) Error() string {
	return e.message
}
func (e *PaperDbException) Unwrap() error {
	return e.cause
}

func TestMessageConstructorPublic(t *testing.T) {
	ex := &PaperDbException{message: "fail-public"}
	if ex.Error() != "fail-public" {
		t.Errorf("Expected message 'fail-public', got %q", ex.Error())
	}
}

func TestMessageAndThrowableConstructorPublic(t *testing.T) {
	tCause := errors.New("public-cause")
	ex := &PaperDbException{message: "fail-public2", cause: tCause}
	if ex.Error() != "fail-public2" {
		t.Errorf("Expected message 'fail-public2', got %q", ex.Error())
	}
	if !errors.Is(ex, tCause) && ex.cause != tCause {
		t.Errorf("Expected cause 'public-cause', got %+v", ex.cause)
	}
}

File: RepoTransBench/target_projects/Java/C#/pilgr_Paper/public_tests/PaperDbExceptionPublicTest.cs
using System;
using Xunit;

namespace PilgrPaper.PublicTests
{
    public class PaperDbExceptionPublicTest
    {
        [Fact]
        public void TestMessageConstructorPublic()
        {
            var ex = new PaperDbException("fail-public");
            Assert.Equal("fail-public", ex.Message);
        }

        [Fact]
        public void TestMessageAndThrowableConstructorPublic()
        {
            var t = new Exception("public-cause");
            var ex = new PaperDbException("fail-public2", t);
            Assert.Equal("fail-public2", ex.Message);
            Assert.Equal(t, ex.InnerException);
        }
    }
}